tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -826,6 +827,583 @@ TEST_F( EnergyPlusFixture, EMSManager_CheckIfAnyEMS_OutEMS ) {
 
 }
 
+TEST_F( EnergyPlusFixture, EMSManager_TestFuntionCall ) {
+		// test to demonstrate accurate function calls.
 
+		std::string const idf_objects = delimited_string( {
+
+			"Curve:Quadratic,",
+			"  TestCurve,       !- Name",
+			"  0.8,             !- Coefficient1 Constant",
+			"  0.2,             !- Coefficient2 x",
+			"  0.0,             !- Coefficient3 x**2",
+			"  0.5,             !- Minimum Value of x",
+			"  1.5;             !- Maximum Value of x",
+
+			"EnergyManagementSystem:ProgramCallingManager,",
+			"Test inside HVAC system iteration Loop,",
+			"InsideHVACSystemIterationLoop,",
+			"Test_InsideHVACSystemIterationLoop;",
+
+			"EnergyManagementSystem:Program,",
+			"Test_InsideHVACSystemIterationLoop,",
+			"set Var1 = @Round 2.1,",
+			"set Var2 = @Mod 7 3,",
+			"set Var3 = @Sin 0.785398163,", // 45 degrees
+			"set Var4 = @Cos 0.785398163,", // 45 degrees
+			"set Var5 = @ArcCos 0.70710678090550916,",
+			"set Var6 = @ArcSin 0.70710678090550916,",
+			"set Var7 = @DegToRad 45.0,",
+			"set Var8 = @RadToDeg 0.785398163,",
+			"set Var9 = @Exp 1.0,",
+			"set Var10 = @Ln 2.0,",
+			"set Var11 = @Max 0.5 1.5,",
+			"set Var12 = @Min 0.5 1.5,",
+			"set Var13 = @Abs 1.3,",  // -3.1 does not work, (-3.1) does not work, "set absNum = -3.1" and use @Abs absNum does work if EMS line of code occurs before this line
+			"set Var14 = @SEEDRANDOM 2.65,",
+			"set Var15 = @RANDOMUNIFORM 0.0 1.0,",
```

Comment:
I think the behavior of these random functions could vary on different hardware.  I don't really know how the C++ is implementing this but it used to be that the fortran called the processor's random generator and behavior of that would vary with the type of processor hardware.  I do not think these are going to be stable enough for unit testing in general and they cause test failures in CI now.   So I think these three random function tests should be removed.